qgis_hub_plugin/gui/resource_browser.py

```diff
@@ -238,3 +262,41 @@ def __init__(self, params: dict):
             self.setIcon(get_icon("qbrowser_icon.svg"))
 
         self.setData(self.resource_type, self.ResourceTypeRole)
+        self.setData(self.name, self.NameRole)
+        self.setData(self.creator, self.CreatorRole)
+
+
+class MultiRoleFilterProxyModel(QSortFilterProxyModel):
+    def __init__(self, parent=None):
+        super().__init__(parent)
+        self.roles_to_filter = []
+        self.checkbox_states = {}
+
+    def setRolesToFilter(self, roles):
+        self.roles_to_filter = roles
+        self.invalidateFilter()
+
+    def setCheckboxStates(self, checkbox_states):
+        self.checkbox_states = checkbox_states
+        self.invalidateFilter()
+
+    def filterAcceptsRow(self, source_row, source_parent):
+        model = self.sourceModel()
+        if not self.roles_to_filter or not self.checkbox_states:
+            return True
+
+        index = model.index(source_row, 0, source_parent)
+        resource_type = model.data(index, ResourceItem.ResourceTypeRole)
+        if resource_type not in self.checkbox_states:
+            return False
+
+        checkbox_checked = self.checkbox_states[resource_type]
+        if not checkbox_checked:
+            return False
```

Comment:
You can simplify these two `if`s with something like this:
```python
if not self.checkbox_states.get(resource_type, False):
   return False
```

Reply:
Implemented it in the latest commits